2015/presentations/Dokku_Build_your_own_PaaS/text.md

```diff
@@ -0,0 +1,128 @@
+but it is possible to avoid those expenses. And now is time to say one word: “Dokku”.
+
+## Definitions
+**Platform as a service (PaaS)** is a category of cloud computing services that provides a platform allowing customers
+to develop, run, and manage Web applications without the complexity of building and maintaining the infrastructure
+typically associated with developing and launching an app. PaaS can be delivered in two ways: as a public cloud
+service from a provider, where the consumer controls software deployment and configuration settings, and the provider
+provides the networks, servers, storage and other services to host the consumer's application; or as software
+installed in private data centers or public infrastructure as a service and managed by internal IT departments.
+
+**Heroku** is a cloud platform as a service (PaaS) supporting several programming languages.
+
+**Dokku** is a Docker-powered PaaS implementation.
+
+## PaaS use case
+Let'ts imagine that you have a small blog written in Django which you need to deploy fast an easy. Also this blog is
+under development and you are going to roll out updates every day during the upcoming month. Simultaneously, lets
+imagine default actions which you do when you need to match such requirements. You purchase a new virtual or
+dedicated server, or buy a new EC2 instance, set up a new environment for the project, or adopt your existing
+infrastructure to requirements of your new blog, what can be especially painful if you have projects with other
+system level requirements which can even use different languages. If you manage with very diverse technologies
+stacks, sooner or later, your server becomes a trash can. But this is still not all pain which you experience in
+such kind routine. If your changes to the code base require system configuration or packages modifications, you start
+spending more time on deployment than on development. This is the right time to say: "Stop! I'm not a system
+administrator! I don't want to configure and maintain, I want to write code!". And right after this phrase you become
+a target audience for PaaS providers.
+
+## Installing Dokku
+Let's assume that you already have configured and running server where we will install Dokku.
+SSH into your host machine and run the following command:
+```
+server $ wget -qO- https://raw.github.com/progrium/dokku/v0.3.26/bootstrap.sh | sudo DOKKU_TAG=v0.3.26 bash
+```
+Please, **notice** that I use the latest version of Dokku (0.3.26) for this moment.
+Also, I want to emphasize that you have to run installation using **sudo** even if you are logged in via root user. This is important as this instalaltor creates a custom dokku user for executing commands and it is important to make sure if this user has enough permissions for building packages, creating configs and required directories. Otherwise your future project builds will fail.

```

Comment:
If that's true, then sth. is totally f*** up there, imho. But ok, I'll revert that change.